aten/src/ATen/test/tensor_iterator_test.cpp

```diff
@@ -105,3 +105,49 @@ TEST(TensorIteratorTest, SerialLoopSingleThread) {
   });
 }
 
+TEST(TensorIteratorTest, InputDType) {
```

Comment:
Please add tests when tensor is input and output at same time.

Reply:
@VitalyFedyunin what do you think we should do in such case?

Reply:
We should not change types for sure, because it is clear case of in-place operations. As for tests I just want to make sure we test cover all expected behavior.